jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/MatchingAlgorithm.java

```diff
@@ -17,24 +17,63 @@
  */
 package org.jgrapht.alg.interfaces;
 
-import java.util.*;
+import java.util.Set;
 
 /**
- * Allows to derive <a href="http://en.wikipedia.org/wiki/Matching_(graph_theory)">matching</a> from
- * given graph
+ * Allows to derive a <a href="http://en.wikipedia.org/wiki/Matching_(graph_theory)">matching</a> of
+ * a given graph.
  *
  * @param <V> the graph vertex type
  * @param <E> the graph edge type
- * 
  */
 public interface MatchingAlgorithm<V, E>
 {
     /**
+     * Default tolerance used by algorithms comparing floating point values.
+     */
+    double DEFAULT_EPSILON = 1e-9;
+
+    /**
      * Returns set of edges making up the matching
      * 
      * @return a matching
+     * @deprecated Use {@link #computeMatching()} instead.
      */
-    Set<E> getMatching();
+    @Deprecated
+    default Set<E> getMatching()
+    {
+        return computeMatching().getEdges();
+    }
+
+    /**
+     * Compute a matching for a given graph.
+     * 
+     * @return a matching
+     */
+    Matching<E> computeMatching();
+
+    /**
+     * A graph matching.
+     *
+     * @param <E> the graph edge type
+     */
+    interface Matching<E>
```

Comment:
Consider adding a default implementation as was done with MaximumFlow:
MatchingImpl extends Matching.
I think @jsichi started this convention for the flow implementations.
If you add a default implementation, please add a toString() method as well.